src/libraries/System.Net.WebProxy/src/System/Net/WebProxy.cs

```diff
@@ -219,7 +224,7 @@ private sealed class ChangeTrackingArrayList : ArrayList
 
             public ChangeTrackingArrayList(ICollection c) : base(c) { }
 
-            public bool IsChanged { get; set; }
+            public volatile bool IsChanged;
```

Comment:
I think it'd be worth adding a comment as to why this is volatile, since that's a bit odd in a type that's not intended to be used concurrently, e.g.
```suggestion
            // While this type isn't intended to mutated concurrently with reads, non-concurrent updates
            // to the list might result in lazy initialization, and it's possible concurrent requests could race
            // to trigger that initialization.
            public volatile bool IsChanged;
```